Source/Core/Core/HW/EXI/BBA/TAPServer_Apple.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fcntl.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Core/HW/EXI/EXI_Device.h"
+#include "Core/HW/EXI/EXI_DeviceEthernet.h"
+
+namespace ExpansionInterface
+{
+// This interface is only implemented on macOS, since macOS needs a replacement
+// for TunTap when the kernel extension is no longer supported. This interface
+// only appears in the menu on macOS, so on other platforms, it does nothing and
+// refuses to activate.
+
+static const std::string socket_path = "/tmp/dolphin-tap";
+
+bool CEXIETHERNET::TAPServerNetworkInterface::Activate()
+{
+  if (IsActivated())
+    return true;
+
+  struct sockaddr_un sun;
+  memset(&sun, 0, sizeof(sun));
+  if ((socket_path.size() + 1) > sizeof(sun.sun_path))
+  {
+    ERROR_LOG(SP1, "Socket path is too long, unable to init BBA");
+    return false;
+  }
+  sun.sun_family = AF_UNIX;
+  strcpy(sun.sun_path, socket_path.c_str());
+
+  fd = socket(AF_UNIX, SOCK_STREAM, 0);
+  if (fd == -1)
+  {
+    ERROR_LOG(SP1, "Couldn't create socket, unable to init BBA");
+    return false;
+  }
+
+  if (connect(fd, (struct sockaddr*)&sun, sizeof(sun)) == -1)
+  {
+    ERROR_LOG(SP1, "Couldn't connect socket (%d), unable to init BBA", errno);
+    close(fd);
+    fd = -1;
+    return false;
+  }
+
+  INFO_LOG(SP1, "BBA initialized.\n");
+  return RecvInit();
+}
+
+bool CEXIETHERNET::TAPServerNetworkInterface::SendFrame(const u8* frame, u32 size)
+{
+  {
+    const std::string s = ArrayToString(frame, size, 0x10);
+    INFO_LOG(SP1, "SendFrame %x\n%s\n", size, s.c_str());
+  }
+
+  uint16_t size16 = size;
```

Comment:
```suggestion
  auto size16 = u16(size);
```

otherwise this can cause a truncation warning.